crates/types/src/logs/metadata.rs

```diff
@@ -154,6 +156,20 @@ pub enum ProviderKind {
     Replicated,
 }
 
+impl FromStr for ProviderKind {
```

Comment:
Perhaps what you want is to derive ValueEnum. Check nodes_config.rs to see an example.

Reply:
ValueEnum is really useful! thanks. But can we also keep the FromStr because it's easy to convert from Protobuf `ProviderKind` to `metadata:ProviderKind`

Reply:
but FromStr is also failable. And we actually do need it since now the `provider` kind is sent as `string` in the protobuf message